packages/grafana-ui/rollup.config.ts

```diff
@@ -35,10 +35,17 @@ const buildCjsPackage = ({ env }) => {
       'moment',
       'jquery', // required to use jquery.plot, which is assigned externally
       'react-inlinesvg', // required to mock Icon svg loading in tests
+      'uplot', // load uplot library dynamically
+      '@emotion/react',
+      '@emotion/css',
+      '@emotion/cache',
+      '@emotion/babel-plugin',
+      '@emotion/utils',
```

Comment:
This is a pretty big breaking change, so this cannot be backported to version before grafana 9. Just an FYI.

Random comments:
 - Not _totally_ sure about the implication of some of these, but on first glance it seems fine to do. 
 - Looks like grafana-ui only declares `@emotion/css` and `@emotion/react` in it's package.json? 
 - Because the consuming project must then supply the emotion libraries, perhaps these here should be moved to `peerDependencies` instead?
    - This would then require the new plugin templates to be updated to make sure they satisfy this

Either way - this breaking change and the implications should be documented

Reply:
I understand what you're saying, and it seems to make sense, but now it's revealed just how little I understand exactly how these libraries are used and bundled for external projects/plugins (and in grafana itself!).

Maybe @grafana/plugins-platform-frontend could provide some insight/advice here? 😇

Reply:
> This is a pretty big breaking change, so this cannot be backported to version before grafana 9. Just an FYI.

I don't believe this is a breaking change either. As @jalopez points out `@grafana/ui` already declares these as dependencies so when `npm install` is run it will pull down all the dependencies and the nested dependencies of those dependencies. The externalised dependencies will be resolved with `var React = require('react');` within the bundle rather than the entire library existing in the bundled `@grafana/ui`.

> it's revealed just how little I understand exactly how these libraries are used and bundled for external projects/plugins (and in grafana itself!)

Currently our NPM packages are workspaces to Grafana core. When running webpack to build Grafana it is the `@grafana/ui` source code which is resolved via `"main": "src/index.ts"` in `package.json`. However when the rollup bundling occurs the `package.json` is copied and the main entry is replaced with `"main": "index.js",`. Now any consumer of the `@grafana/ui` outside Grafana core will be working with the bundled code. It's probably worth noting that due to the entire workspace being consumed by `plugin_loader.ts` webpack isn't tree-shaking any of `@grafana/ui` making sure plugin devs have access to entire library at runtime.

 > Actually, the bundle should only include the lib code and not any dependency,
 
I agree and want to revisit the rollup configs and how Grafana consumes the `@grafana` packages to better address the differences in how these workspaces / packages are used by Grafana core and Plugin devs / other projects. I hope this explains things @joshhunt .

Reply:
Thanks super useful!